crates/store/src/server/api.rs

```diff
@@ -258,76 +257,22 @@ impl api_server::Api for StoreApi {
-
-        let nullifiers = validate_nullifiers(&request.nullifiers)?;
-        let accounts = request
-            .accounts
-            .iter()
-            .map(|account_update| {
-                let account_state: AccountState = account_update
-                    .try_into()
-                    .map_err(|err: ConversionError| Status::invalid_argument(err.to_string()))?;
-
-                match (account_state.account_id.is_on_chain(), account_update.details.is_some()) {
-                    (true, false) => {
-                        return Err(Status::invalid_argument("On-chain account must have details"));
-                    },
-                    (false, true) => {
-                        return Err(Status::invalid_argument(
-                            "Off-chain account must not have details",
-                        ));
-                    },
-                    _ => (),
-                }
-
-                let details = account_update
-                    .details
-                    .as_ref()
-                    .map(|data| AccountDetails::read_from_bytes(data))
-                    .transpose()
-                    .map_err(|err| Status::invalid_argument(err.to_string()))?;
-
-                Ok(AccountUpdateDetails {
-                    account_id: account_state.account_id,
-                    details,
-                    final_state_hash: account_state
-                        .account_hash
-                        .ok_or(invalid_argument("Account update missing account hash"))?,
-                })
-            })
-            .collect::<Result<Vec<_>, Status>>()?;
-
-        let notes = request
-            .notes
-            .into_iter()
-            .map(|note| {
-                Ok(NoteCreated {
-                    batch_index: note.batch_index,
-                    note_index: note.note_index,
-                    note_id: note
-                        .note_id
-                        .ok_or(invalid_argument("Note missing id"))?
-                        .try_into()
-                        .map_err(|err: ConversionError| {
-                            Status::invalid_argument(err.to_string())
-                        })?,
-                    note_type: NoteType::try_from(note.note_type as u64)
-                        .map_err(|err: NoteError| Status::invalid_argument(err.to_string()))?,
-                    sender: note.sender.ok_or(invalid_argument("Note missing sender"))?.into(),
-                    tag: note.tag,
-                    details: note.details,
-                })
-            })
-            .collect::<Result<Vec<_>, Status>>()?;
-
-        let _ = self.state.apply_block(block_header, nullifiers, accounts, notes).await;
+        let block = Block::read_from_bytes(&request.block).map_err(|err| {
+            Status::invalid_argument(format!("Block deserialization error: {err}"))
+        })?;
```

Comment:
Question: do we still get all the same validations as we had here before? (I think we do, but wanted to double-check).

Reply:
Yes, we check it during validation of proven transaction: https://github.com/0xPolygonMiden/miden-base/blob/a01390c8fc7dd5cb4937d58e17fea665b5b4d9e8/objects/src/transaction/proven_tx.rs#L84-L99

Reply:
Do we not also do these checks during block deserialization?

Reply:
We don't need it, since this endpoint is not accessible outside and block-producer does this validation before block building.

Reply:
I think it would still be good to make sure the block is well-formed here - but I also think this is already happening. Are there any checks that we did in the removed code which we are not doing any more?